src/map/options.ts

```diff
@@ -243,6 +244,9 @@ export class MapOptions extends OptionsGroup {
                         scaled = 1.0 / scaled;
                         break;
                     case 'log':
+                        if (scaled < 0) {
+                            sendWarning("This property contains negative values, taking the log will discard them.");
```

Comment:
This will send multiple warnings if the property contains multiple negative values. I think it would be better to move this check out of the `rawSizes.map` call; and only warn once.